platform/darwin/src/MGLShapeSource.mm

```diff
@@ -56,6 +56,16 @@ - (instancetype)initWithIdentifier:(NSString *)identifier shape:(nullable MGLSha
     return self;
 }
 
+- (instancetype)initWithIdentifier:(NSString *)identifier features:(NS_ARRAY_OF(MGLFeature) *)features options:(nullable NS_DICTIONARY_OF(MGLShapeSourceOption, id) *)options {
```

Comment:
`NS_ARRAY_OF(MGLFeature) *` resolves to `NSArray<MGLFeature> *`, meaning the passed-in value must be an NSArray object that conforms to the MGLFeature protocol. I think what you meant was `NS_ARRAY_OF(id <MGLFeature>) *`.